### PR TITLE
fix wss /wails/ipc

### PR DESCRIPTION
**Pull request by zuiwuchang**
Fix `wails dev` cannot access websocket `wails/ipc` via wss #2436

## Summary by CodeRabbit

- **Bug Fixes**
	- Updated WebSocket connection to support both HTTP and HTTPS protocols, enhancing compatibility and security.

### CODE REVIEW

**Comment by leaanthony** [comment]
Thanks for submitting this! @stffabi - thoughts?

File: v2/internal/frontend/runtime/dev/main.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ function handleDisconnect() {
 
 function _connect() {
     if (websocket == null) {
-        websocket = new WebSocket('ws://' + window.location.host + '/wails/ipc');
+        websocket = new WebSocket((window.location.protocol.startsWith("https") ? "wss://" : "ws://") + window.location.host + "/wails/ipc");
         websocket.onopen = handleConnect;
```
Comment (tmclane) [inline v2/internal/frontend/runtime/dev/main.js:84]: Maybe we should move this logic into a function that simply returns the connection string.  Would clean up this particular location in the code.   Presumably it only impacts 'dev' and 'server' type modes.

**Comment by learncodingforweb** [comment]
i am also facing same issue on ubuntu ``#47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 21 15:35:31 UTC 2``
**wails doctor ouput is**
```
Wails CLI v2.5.1

 SUCCESS  Done.                                                                                              

# System

OS           | Ubuntu  
Version      | 22.04   
ID           | ubuntu  
Go Version   | go1.20.5
Platform     | linux   
Architecture | amd64   

# Wails

Version         | v2.5.1
Package Manager | apt   

# Dependencies

Dependency | Package Name          | Status    | Version                
*docker    | docker.io             | Installed | 24.0.4                 
gcc        | build-essential       | Installed | 12.9ubuntu3            
libgtk-3   | libgtk-3-dev          | Installed | 3.24.33-1ubuntu2       
libwebkit  | libwebkit2gtk-4.0-dev | Installed | 2.38.6-0ubuntu0.22.04.1
npm        | npm                   | Installed | 8.19.2                 
*nsis      | nsis                  | Installed | v3.08-2                
pkg-config | pkg-config            | Installed | 0.29.2-1ubuntu3        
* - Optional Dependency
```

when try to connect socket io 
const sock = io("ws://localhost:8000", {path: '/web/socket.io/'});

Solution tried, if try to run on browser like chrome and mozilla, it is working fine. but on wails app getting 
``WebSocket connection to 'wss://wails.localhost:34115/_next/webpack-hmr' failed: ``

File: v2/internal/frontend/runtime/dev/main.js (hunk 1 above)
Comment (ollytom) [reply v2/internal/frontend/runtime/dev/main.js:84]: Could drop the ternary operator to keep the line short:

```
let proto = 'ws:';
if (window.location.protocol == 'https:') {
    proto = 'wss:';
}
websocket = new WebSocket(proto + '//' + window.location.host + '/wails/ipc');
```

**Comment by munjalpatel** [comment]
I am also have similar issues trying to run `next` hot-reload with Wails.

**Error:**
```
WebSocket connection to 'wss://wails.localhost:34115/_next/webpack-hmr' failed: An SSL error has occurred and a secure connection to the server cannot be made.
```

**Network Call Details:**
```
Summary
URL: wss://wails.localhost:34115/_next/webpack-hmr
Status: —
Source: —

Request
Origin: wails://wails.localhost:34115
Pragma: no-cache
Sec-WebSocket-Key: mliiNr+I5PLFnpXwUMznaQ==
Sec-WebSocket-Version: 13
Upgrade: websocket
Sec-WebSocket-Extensions: permessage-deflate
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) wails.io
Cache-Control: no-cache
Connection: Upgrade

Response
No response headers
```

**Comment by xudaolong** [comment]
any update?

**Review by tmclane — APPROVED** [review]

**Pull request merged**